listenbrainz/webserver/static/css/rc-slider.less

```diff
@@ -0,0 +1,298 @@
+// Converted rc-slider/assets/index.css to .less
+
+// TODO name colors properly
+@color_1: #999;
+@color_2: #666;
+@slider_base_color: #8d8d8d;
+@background_color_2: #abe2fb;
+@background_color_4: #ccc;
+@background_color_5: #6c6c6c;
+@border_color_1: @orange;
+@border_color_2: @orange;
+@border_color_3: #96dbfa;
+@border_color_4: #ccc;
+@border_color_5: transparent;
+@border_top_color_1: #6c6c6c;
+
+@keyframes rcSliderTooltipZoomDownIn {
+  0% {
+    transform: scale(0, 0);
+    transform-origin: 50% 100%;
+    opacity: 0;
+  }
+  100% {
+    transform: scale(1, 1);
+    transform-origin: 50% 100%;
+  }
+}
+@keyframes rcSliderTooltipZoomDownOut {
+  0% {
+    transform: scale(1, 1);
+    transform-origin: 50% 100%;
+  }
+  100% {
+    transform: scale(0, 0);
+    transform-origin: 50% 100%;
+    opacity: 0;
+  }
+}
+.rc-slider {
+  position: relative;
+  width: 100%;
+  height: 14px;
+  padding: 5px 0;
+  border-radius: 6px;
+  touch-action: none;
+  box-sizing: border-box;
+  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+  * {
+    box-sizing: border-box;
+    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+  }
+}
+.rc-slider-rail {
+  position: absolute;
+  width: 100%;
+  height: 6px;
+  background-color: @slider_base_color;
+  border-radius: 6px;
+
+  &:hover ~ .rc-slider-handle {
+    border: solid 5px lighten(@orange, 10%);
+  }
+}
+.rc-slider-track {
+  position: absolute;
+  height: 6px;
+  background-color: @background_color_2;
+  border-radius: 6px;
+}
+.rc-slider-handle {
+  position: absolute;
+  width: 3rem;
+  height: 3rem;
+  margin-top: -12px;
+  background-color: @orange;
+  border-radius: 50%;
+  cursor: pointer;
+  cursor: -webkit-grab;
+  cursor: grab;
+  touch-action: pan-x;
+  transition: all 0.5s cubic-bezier(0.46, 0.03, 0.52, 0.96) 0s;
+  &:focus {
+    outline: none;
+    box-shadow: none;
+  }
+  &:focus-visible {
+    border-color: lighten(@purple, 10%);
+    box-shadow: 0 0 3px @purple;
+  }
+  &:hover {
+    border: solid 5px lighten(@orange, 10%);
+  }
+  &:active {
+    border-color: @purple;
+    box-shadow: 0 0 8px darken(@purple, 10%);
+    cursor: -webkit-grabbing;
+    cursor: grabbing;
+  }
+}
+.rc-slider-handle-dragging.rc-slider-handle-dragging.rc-slider-handle-dragging {
+  background-color: @purple;
+  border-color: lighten(@purple, 10%);
+  box-shadow: 0 0 7px @purple;
+}
+.rc-slider-handle-click-focused {
+  &:focus {
+    border-color: @orange;
+    box-shadow: unset;
+  }
+}
+.rc-slider-mark {
+  position: absolute;
+  top: 18px;
+  left: 0;
+  width: 100%;
+  font-size: 12px;
+}
+.rc-slider-mark-text {
```

Comment:
I like the css transform version better than the odd/even approach.

Now that we're on this topic I have this whole idea for marks and the handle logic. It may take some time to figure it out. The idea goes like this -
Have the marks placed equidistantly no matter what how many releases are per day. The handle will scroll to the positon where the marks are while actually the grid is getting scrolled to the correct position i.e. the respective date. This way the timeline looks more polished and easier to navigate to the wanted date. As the main purpose of having the timeline is to have something like a map legend to make it easily navigable between dates. The placement of dates doesn't matter as long as they are there (in a visually appealing way). We can leverage the `steps` property of rc-slider but we'll have to figure out the logic.

But for now I'll go with css transform to declutter them.

Reply:
Update: Rotated those marks!